test/rnn_seq_api.hpp

```diff
@@ -0,0 +1,1663 @@
+                                       dropout_rate,
+                                       dropout_state_buf,
+                                       statesSizeInBytes,
+                                       dropout_seed,
+                                       false,
+                                       false,
+                                       MIOPEN_RNG_PSEUDO_XORWOW);
+
+            miopenSetRNNDescriptor_V2(&rnnDesc,
+                                      hiddenSize,
+                                      numLayers,
+                                      &dropoutDesc,
+                                      miopenRNNInputMode_t(inputMode),
+                                      miopenRNNDirectionMode_t(dirMode),
+                                      miopenRNNMode_t(rnnMode),
+                                      miopenRNNBiasMode_t(biasMode),
+                                      miopenRNNAlgo_t(algoMode),
+                                      type);
+        }
+        else
+        {
+            miopenSetRNNDescriptor(&rnnDesc,
+                                   hiddenSize,
+                                   numLayers,
+                                   miopenRNNInputMode_t(inputMode),
+                                   miopenRNNDirectionMode_t(dirMode),
+                                   miopenRNNMode_t(rnnMode),
+                                   miopenRNNBiasMode_t(biasMode),
+                                   miopenRNNAlgo_t(algoMode),
+                                   type);
+        }
+
+        const auto inOut_layout   = rnn_data_layout(io_layout);
+        const auto out_vector_len = hiddenSize * ((dirMode != 0) ? 2 : 1);
+
+        T padding_m = static_cast<T>(0);
+
+        seqTensor<T> input = build_RNN_seqTensor<T>(
+                         inOut_layout, batchSize, seqLength, inVecLen, seqLenArray, &padding_m),
+                     output = build_RNN_seqTensor<T>(inOut_layout,
+                                                     batchSize,
+                                                     seqLength,
+                                                     out_vector_len,
+                                                     seqLenArray,
+                                                     &padding_m);
+        seqTensor<T> dy(output);
+
+        const auto num_hidden_layers = numLayers * ((dirMode != 0) ? 2 : 1);
+        tensor<T> hx(std::vector{num_hidden_layers, batchSize, hiddenSize});
+        tensor<T> cx(hx), dhy(hx), dcy(hx);
+
+        std::vector<T> weights(get_RNN_params_byteSize(handle, rnnDesc, input.desc) / sizeof(T));
+
+        check_GPU_mem_limit(
+            handle, rnnDesc, input, output, hx, cx, weights.size() * sizeof(T), statesSizeInBytes);
+
+        fill_buffers(input, dy, hx, cx, dhy, dcy, weights);
+
+        // avoid BWD unexpected fails
+        if(inVecLen == 1 && hiddenSize == 13 && seqLength == 1 && batchSize == 1)
+        {
+            tolerance = 110;
+        }
```

Comment:
[Q] This looks like a W/A for some problem. Does it worth a ticket?

Reply:
No this is WA for our data generator and our comparing function. At this combination too much information accumulated in single element and his value was around float::eps(), 
std::sqrt(square_difference) / (std::sqrt(n) * mag) => std::sqrt(square_diff(cpu,gpu)) / (1 * max(cpu,gpu))
so if we have single element in container this is now relative difference of single value, but we still using eps as threshold.
as example:
CPU = 5.960e-06, GPU = 5.9606e-06 -> std::sqrt(square_diff(cpu,gpu)) =6.0e-10; 
6.0e-10/(1 * 5.9606e-06)=1.0e4 .
And 1.0e4 bigger than eps*80 so this test failed

Reply:
[Recommendation] I would add an explanatory comment (the explanation above seems perfect). Maybe it's even worth changing the tolerance from 80 to 160 (regardless of the rnn parameters) and removing the conditional. But I am not sure, especially about the latter. Up to you.

[repolved]